### PR TITLE
fix(platform-value): reject lossy CBOR integer conversions

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## What

- replace unchecked 128-bit CBOR integer narrowing with checked native-range conversion
- preserve the full native CBOR signed and unsigned integer range
- reject out-of-range values before nested maps or arrays can emit altered data
- add boundary, round-trip, collision-prevention, and nested-container regressions

## Why

Distinct Platform Value integers could previously serialize to the same CBOR integer, and some native CBOR values could change sign during a decode/re-encode cycle. The conversion now fails closed with the existing IntegerSizeError.

This covers PLATFORM-DS-CAND-366 and PLATFORM-DS-CAND-367.

## Checks

- cargo fmt --all -- --check
- cargo test -p platform-value --features cbor --lib --tests (1,316 passed)
- cargo clippy -p platform-value --features cbor --all-targets -- -D warnings

The complete package command also ran all unit and integration tests successfully; its unrelated pre-existing doctest example fails because serde_json is not available to that doctest target.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved CBOR conversion for 128-bit signed and unsigned values by using safe integer conversion instead of narrowing.
  * Out-of-range 128-bit values now reliably fail with a dedicated integer size error.
  * Nested composite values containing out-of-range 128-bit integers correctly cause the overall conversion to fail.
* **Tests**
  * Expanded test coverage for boundary behavior and verified CBOR serialization/deserialization round-trips for signed 128-bit values within the native range.

### CODE REVIEW

**Comment by thepastaclaw** [comment]
✅ Final review complete — no blockers (commit 9ce1e946)

File: packages/rs-platform-value/src/converter/ciborium.rs [hunk 1]
```diff
@@ -535,6 +593,23 @@ mod tests {
         assert!(!buf.is_empty());
     }
 
+    #[test]
+    fn to_cbor_buffer_roundtrips_native_cbor_integer_boundaries() {
+        const MIN_NATIVE_CBOR_INTEGER: i128 = -(u64::MAX as i128) - 1;
```
Comment (thepastaclaw) [inline packages/rs-platform-value/src/converter/ciborium.rs]: **💬 Nitpick: MIN_NATIVE_CBOR_INTEGER boundary constant duplicated across three test functions**

`const MIN_NATIVE_CBOR_INTEGER: i128 = -(u64::MAX as i128) - 1;` is independently redefined inside `i128_preserves_the_full_native_cbor_integer_range` (line 400), `i128_rejects_values_outside_the_native_cbor_integer_range` (line 418), and `to_cbor_buffer_roundtrips_native_cbor_integer_boundaries` (line 598). This is the exact boundary value the whole PR is about, so keeping it in three places means a future edit could update it in one test but not the others, silently weakening coverage of the edge case this PR was written to protect.

```suggestion
// at module scope, e.g. just inside `mod tests {`
const MIN_NATIVE_CBOR_INTEGER: i128 = -(u64::MAX as i128) - 1;
```

<sub>source: ['claude']</sub>

**Review by thepastaclaw — COMMENTED** [review]
## Final validation — Codex + Sonnet

This PR fixes a real correctness bug where U128/I128 platform values outside ciborium's native CBOR integer range were silently truncated via `as u64`/`as i64` casts, causing distinct values to collide or flip sign on decode. The fix replaces these with checked `Integer::try_from` conversions that I verified against ciborium 0.2.2's actual TryFrom<i128>/TryFrom<u128> source match the native range exactly, failing closed with the crate's existing `Error::IntegerSizeError` pattern. All 38 relevant tests pass, clippy is clean with warnings denied, and the new boundary/rejection/nested-container/round-trip tests correctly exercise the fix. Only a minor test-code duplication nitpick was found; no blocking issues.

### Review provenance
- Codex reviewers: `gpt-5.6-sol` — general (completed), `gpt-5.6-sol` — security-auditor (completed), `gpt-5.6-sol` — rust-quality (completed)
- Verifier: `claude-sonnet-5` — final-verifier
- Sonnet reviewers: `claude-sonnet-5` — general (completed), `claude-sonnet-5` — security-auditor (completed), `claude-sonnet-5` — rust-quality (completed)

💬 1 nitpick(s)

**Review by thepastaclaw — APPROVED** [review]
## Final validation — Codex + Sonnet

This delta is a pure test-only cleanup: it hoists the previously triplicated MIN_NATIVE_CBOR_INTEGER constant to module scope in packages/rs-platform-value/src/converter/ciborium.rs and updates the three boundary/rejection/round-trip tests to reference it. No production code, validation logic, or consensus-relevant behavior changed. Both Codex and Claude reviewer lanes independently confirm the prior duplication finding is fixed and report no new defects.

### Review provenance
- Codex reviewers: `gpt-5.6-sol` — general (failed), `gpt-5.6-sol` — rust-quality (failed), `gpt-5.6-sol` — general (failed), `gpt-5.6-sol` — rust-quality (failed), `gpt-5.6-sol` — general (failed), `gpt-5.6-sol` — rust-quality (failed), `gpt-5.6-sol` — general (failed), `gpt-5.6-sol` — rust-quality (failed), `gpt-5.6-sol` — general (failed), `gpt-5.6-sol` — rust-quality (failed), `gpt-5.6-sol` — general (failed), `gpt-5.6-sol` — general (completed), `gpt-5.6-sol` — rust-quality (completed)
- Verifier: `claude-sonnet-5` — final-verifier
- Sonnet reviewers: `claude-sonnet-5` — general (failed), `claude-sonnet-5` — rust-quality (failed), `claude-sonnet-5` — general (failed), `claude-sonnet-5` — rust-quality (failed), `claude-sonnet-5` — general (failed), `claude-sonnet-5` — rust-quality (failed), `claude-sonnet-5` — general (failed), `claude-sonnet-5` — rust-quality (failed), `claude-sonnet-5` — general (completed), `claude-sonnet-5` — rust-quality (completed)

File: packages/rs-platform-value/src/converter/ciborium.rs (hunk 1 above)
Comment (thepastaclaw) [reply packages/rs-platform-value/src/converter/ciborium.rs]: Resolved in `9ce1e94` — _MIN_NATIVE_CBOR_INTEGER boundary constant duplicated across three test functions_ no longer present.

<sub>Auto-resolved by the review system based on the latest commit diff. If you believe this was closed in error, reopen the thread.</sub>
